vb-fe/src/components/select/Select.tsx

```diff
@@ -0,0 +1,99 @@
+/* Copyright (C) 2021 Vibey - All Rights Reserved */
+
+import React, { useEffect, useState, useRef, useCallback } from "react";
+
+import CreatableSelect from "react-select/creatable";
+
+import { Tag } from "util/Tags";
+import { SelectOption, NoSelectOption } from "components/select/SelectOption";
+import { searchTags } from "services/Tag";
+import { customStyles } from "components/select/SelectStyles";
+
+import "components/select/Select.scss";
+
+/**
+ * @function updateTags function emitted that contains the updated tags
+ */
+interface State {
+  updateTags: (tag: Tag) => void;
+  presentTags: Tag[];
+  handleError: (error: any) => void;
+}
+
+export const Select = (props: State) => {
+  const { updateTags, presentTags, handleError } = props;
+  const [tags, setTags] = useState<Tag[]>([]);
+  const [inputValue, setInputValue] = useState<string>("");
+  const selectInputRef = useRef<any>();
```

Comment:
I've been having to use refs quite a bit recently.  I've been handling it by initializing the ref with `null`, and typing it with the specific element type.  For example,
`const selectInputRef = useRef<HTMLSelectInput>(null)`

It looks like this ref is actually not used in this file though, so I think we can remove it.

Reply:
it is used right? 

Line 85: ref={selectInputRef}

Reply:
but year its not referenced anywhere i'll remove it